### DISCUSSION TITLE
Understanding permission in new macOS instances

### DISCUSSION THREAD

**Discussion by ericoporto** [cirruslabs/cirrus-ci-docs#1129] [Q&A] [answered]
Hi,

So, I am experimenting the new macOS images, it took me a bit to figure it out what to actually write in the yaml file to get it to switch from Intel to Apple Silicon,to get it running again.

But now I have an error I am not sure what it means

https://cirrus-ci.com/task/5716167344521216

I am trying to get Ninja in `/usr/local/bin/ninja` so I can run it, but I get a permission denied error when trying to copy it. This used to work in the Intel images, is there something I need to do?

Additional question, does the Apple Silicon used by Cirrus comes with the feature to run Intel-64 binaries too?

**Comment by fkorotkov** [comment, marked as answer]
Hey @ericoporto,

My guess is that System Integrity Protection in newer versions of macOS restricted more things like touching `/usr/local/bin` without `sudo`. I'd suggest to try to `sudo cp` insted.

Also if you have an Apple Silicone machine at your hands, you can try to play around with the VMs locally using our virtualization solution called Tart: https://github.com/cirruslabs/tart

**Reply by ericoporto** [reply]
Thank you! I didn't knew it was an OS change, thought something was different in the image making process.
